### PR TITLE
Fix Magnifier crash

### LINKED ISSUE #111301: Crash: "_CastError at SelectionOverlay.showMagnifier"
A crash has been reported related to the loupe/magnifier feature (introduced in https://github.com/flutter/flutter/pull/107477) at this line:

https://github.com/flutter/flutter/blob/d7f6a4dbfb4122e73c60b54717ac0a0a24f3a449/packages/flutter/lib/src/widgets/text_selection.dart#L864

I'm not sure how exactly this is being reproduced in the wild, but it seems the loupe shouldn't assume that the handles are visible here.  Fix incoming!

b/244274164

### PR DESCRIPTION
**Pull request by justinmc**
Fixes a crash that was happening when the magnifier was shown and no handles were visible.

Fixes https://github.com/flutter/flutter/issues/111301

### CODE REVIEW

File: packages/flutter/lib/src/widgets/text_selection.dart [hunk 1]
```diff
@@ -861,7 +861,7 @@ class SelectionOverlay {
         context: context,
         below: magnifierConfiguration.shouldDisplayHandlesInMagnifier
             ? null
-            : _handles!.first,
```
Comment (Renzo-Olivares) [inline packages/flutter/lib/src/widgets/text_selection.dart:864]: Should the `|| _handles == null` above on line 856 be removed?

File: packages/flutter/test/widgets/text_selection_test.dart [hunk 2]
```diff
@@ -1279,6 +1281,28 @@ void main() {
       await tester.pump(const Duration(milliseconds: 20));
       expect(endDragEndDetails, isNotNull);
     });
+
+    testWidgets('can show magnifier when no handles exist', (WidgetTester tester) async {
```
Comment (Renzo-Olivares) [inline packages/flutter/test/widgets/text_selection_test.dart:1285]: Should we also, at the end of this test, confirm the magnifier is showing?

File: packages/flutter/lib/src/widgets/text_selection.dart (hunk 1 above)
Comment (justinmc) [reply packages/flutter/lib/src/widgets/text_selection.dart:864]: Yes thank you!  I think you added that check just before I created this PR and I somehow totally missed it.  Likely your PR has already fixed the crash that I was trying to fix with this PR.  I think this PR is still worthwhile to allow the magnifier to show without the handles being visible though.  Let me know what you think of the code now after my change.

Comment (Renzo-Olivares) [reply packages/flutter/lib/src/widgets/text_selection.dart:864]: Yeah I added that check, and I agree this is still a worthwhile change.

**Review by Renzo-Olivares — APPROVED** [review]
LGTM

**Pull request merged**